cfme/infrastructure/host.py

```diff
@@ -621,16 +622,17 @@ def find_quadicon(host, do_not_navigate=False):
         raise HostNotFound("Host '{}' not found in UI!".format(host))
 
 
-def navigate_and_select_all_hosts(host_names, provider_crud=None):
+def navigate_and_select_all_hosts(host_names, provider=None):
     """ Reduces some redundant code shared between methods """
     if isinstance(host_names, basestring):
         host_names = [host_names]
 
-    if provider_crud:
-        navigate_to(provider_crud, 'ProviderNodes')
+    if provider:
+        navigate_to(provider, 'ProviderNodes')
     else:
         navigate_to(Host, 'All')
+
     if paginator.page_controls_exist():
-        paginator.results_per_page(1000)
```

Comment:
REQUIRED: You have removed row which made cfme show you all objects if there are several pages.
And added row paginator.check_all which does nothing. It just gives you select_all input. btw, you left statement where you select all passed elements.
If I understand correctly,
you need to do something like:

 ```
if paginator.page_controls_exist():
      paginator.results_per_page(1000)
      sel.click(paginator.check_all())
else:
    for host_name in host_names:
          sel.check(Quadicon(host_name, 'host').checkbox())
```